src/ripple/overlay/impl/OverlayImpl.cpp

```diff
@@ -366,18 +359,18 @@ OverlayImpl::makeRedirectResponse (PeerFinder::Slot::ptr const& slot,
 }
 
 std::shared_ptr<Writer>
-OverlayImpl::makeErrorResponse (PeerFinder::Slot::ptr const& slot,
+OverlayImpl::makeErrorResponse (std::shared_ptr<PeerFinder::Slot> const& slot,
     http_request_type const& request,
     address_type remote_address,
     std::string text)
 {
-    boost::beast::http::response<boost::beast::http::string_body> msg;
+    boost::beast::http::response<boost::beast::http::empty_body> msg;
     msg.version(request.version());
     msg.result(boost::beast::http::status::bad_request);
+    msg.reason("Bad Request (" + text + ")");
```

Comment:
According to the docs on `reason` this function is deprecated. Ref:
```

    /** Set the response reason-phrase (deprecated)

        This function sets a custom reason-phrase to a copy of
        the string passed in. Normally it is not necessary to set
        the reason phrase on an outgoing response object; the
        implementation will automatically use the standard reason
        text for the corresponding status code.

        To clear a previously set custom phrase, pass an empty
        string. This will restore the default standard reason text
        based on the status code used when serializing.

        The reason-phrase is obsolete as of rfc7230.

        @param s The string to use for the reason-phrase.

        @note This function is only available when `isRequest == false`.
    */
 ```

Reply:
I understand that the reason phrase is deprecated, but we have no better way to communicate an actual error message (short of adding a custom header) that gets displayed to the user.

This is a legit problem with existing `rippled` instances; if you get disconnected for, say, the clock being too far off, nothing gets logged on your end which can make debugging difficult.